packages/connection-form/src/components/advanced-options-tabs/authentication-tab/authentication-oidc.tsx

```diff
@@ -0,0 +1,169 @@
+  const username = getConnectionStringUsername(connectionStringUrl);
+  const usernameError = errorMessageByFieldName(errors, 'username');
+
+  const handleFieldChanged = useCallback(
+    (key: keyof OIDCOptions, value?: OIDCOptions[keyof OIDCOptions]) => {
+      return updateConnectionFormField({
+        type: 'update-oidc-param',
+        key: key,
+        value,
+      });
+    },
+    [updateConnectionFormField]
+  );
+
+  const hasEnabledDeviceAuthFlow = !!(
+    connectionOptions.oidc?.allowedFlows as AuthFlowType[]
+  )?.includes?.('device-auth');
+
+  return (
+    <>
+      <FormFieldContainer>
+        <TextInput
+          data-testid="connection-oidc-username-input"
+          onChange={({
+            target: { value },
+          }: React.ChangeEvent<HTMLInputElement>) => {
+            return updateConnectionFormField({
+              type: 'update-username',
+              username: value,
+            });
+          }}
+          label="Username"
+          optional
+          value={username || ''}
+          errorMessage={usernameError}
+          state={usernameError ? 'error' : undefined}
+        />
+      </FormFieldContainer>
+      <FormFieldContainer>
+        <Accordion text="OIDC Options">
+          <FormFieldContainer>
+            <TextInput
+              data-testid="connection-oidc-auth-code-flow-redirect-uri-input"
+              onChange={({
+                target: { value },
+              }: React.ChangeEvent<HTMLInputElement>) => {
+                handleFieldChanged('redirectURI', value);
+              }}
+              optional
+              label="Auth Code Flow Redirect URI"
+              value={connectionOptions?.oidc?.redirectURI || ''}
+              description="This value needs to match the configuration of the Identity Provider used by the server."
+            />
+          </FormFieldContainer>
+
+          <FormFieldContainer>
+            <Checkbox
+              onChange={({
+                target: { checked },
+              }: React.ChangeEvent<HTMLInputElement>) => {
+                if (checked) {
+                  return handleFieldChanged('enableUntrustedEndpoints', true);
+                }
+
+                // Unset the allowed hosts.
+                return handleFieldChanged(
+                  'enableUntrustedEndpoints',
+                  undefined
+                );
+              }}
+              data-testid="oidc-allow-untrusted-endpoint-input"
+              id="oidc-allow-untrusted-endpoint-input"
+              label={
+                <>
+                  <Label htmlFor="oidc-allow-untrusted-endpoint-input">
+                    Enable Untrusted Target Endpoint
+                  </Label>
+                  <Description>
+                    Allow connecting when the target endpoint is not in the list
+                    of trusted endpoints &#40;this sets the driver&apos;s
+                    authMechanismProperties&apos; ALLOWED_HOSTS list to *&#41;
```

Comment:
I’m not sure I have a specific good suggestion here, but this sounds a bit ambiguous – what the user should know when they check this box is that they promise that this is a _trusted_ endpoint, rather than relying on MongoDB’s preconfigured list of known trusted endpoints. But “Enable Untrusted Target Endpoint” makes it sound a bit like it would put Compass into a more “secure” mode where it handles connections to potentially untrustworthy endpoints well too, if you know what I mean? Maybe something like this:

```suggestion
                    Consider Target Endpoint Trusted
                  </Label>
                  <Description>
                    Allow connecting when the target endpoint is not in the list
                    of endpoints that are considered trusted by default.
                    Only use this option when connecting to servers that you trust.
```

Reply:
Good thinking - thanks for updating!